paddle/fluid/framework/operator.cc

```diff
@@ -981,6 +981,12 @@ void OperatorWithKernel::RunImpl(const Scope& scope,
       }
     }
   }
+
+  if (transfer_scope && !run_by_executor_ &&
+      (platform::is_cpu_place(kernel_type_->place_))) {
+    std::cout << "delete scope transfer_scope=" << transfer_scope << "\n";
```

Comment:
If a GPU model with this Op runs on CPU, will it fail?

Reply:
I think transferscope cache is only for the case that doing convert from cpu tensor to gpu tensor with gpu kernel, in this case, framework will allocate output tensor inside of gpu, then trigger gpu memory explosion issue.
so maybe checking kernel_type_->place_ is enough, even for TryCreateTransferScope part.

Reply:
> I think transferscope cache is only for the case that doing convert from cpu tensor to gpu tensor with gpu kernel

We have the case "a GPU model with this Op runs on CPU", please see #16981

Reply:
I agree, there is case for op running on CPU with a GPU model. But my understanding here is TryCreateTransferScope() is trying to fix GPU memroy explosion, see comment there:

```
      // In the inference scenerio, the scopes will be reused across the
      // batches, so the `new_scope` here will result in GPU memroy explosion
      // over the  running of operators.
```

Why GPU memroy explosion happens ?  I think it is because data transforming will always ask GPU to allocate memory for output instead of input.  Based on this assumption, I think GPU memroy explosion just happens when input is CPU memory while next kernel need GPU memory, 
so kernel type should be definitely GPU place.

Anyway, we'd better to sync with original author or chunwei to understand  the scenario of that GPU issue.